packages/operator/pkg/repository/connection/vault/connection.go

```diff
@@ -153,6 +153,7 @@ func (vcr *vaultConnRepository) UpdateConnection(conn *connection.Connection) er
 	case err == nil:
 		// If err is not nil, then the connection already exists.
 		existedConn.Spec = conn.Spec
+		existedConn.UpdatedAt = conn.UpdatedAt
```

Comment:
All the logic should be removed from repository. The repository is responsible for only saving an object **as is**. Any required checks and modifications should take place on Service level. 
Currently both service and repository make Get request to Vault which is unnecessary.